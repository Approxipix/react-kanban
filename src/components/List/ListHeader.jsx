import React, { Component } from 'react';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome/index";
import ListActions from './ListActions/ListActions';
import ListTitle from './ListTitle/ListTitle';
import styled from 'styled-components'

const Header = styled.div`
  padding: .5rem .7rem 0rem .5rem;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

const Icon = styled.div`
  cursor: pointer;
`;
Icon.displayName = 'Icon';


class ListHeader extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpened: false,
    }
  }

  toggleOpened = () => {
    this.setState({
      isOpened: !this.state.isOpened
    })
  };

  render() {
    const { dragHandleProps, listId, listTitle, boardId } = this.props;
    const { isOpened } = this.state;
    return (
      <>
        <Header>
          <ListTitle
            dragHandleProps={dragHandleProps}
            listId={listId}
            listTitle={listTitle}
          />
          <Icon onClick={() => this.toggleOpened()}>
            <FontAwesomeIcon icon="ellipsis-h" />
          </Icon>
        </Header>
        {isOpened && (
          <ListActions
            toggleOpened={this.toggleOpened}
            boardId={boardId}
            listId={listId}
          />
        )}
      </>
    )
  }
}

export default ListHeader;
