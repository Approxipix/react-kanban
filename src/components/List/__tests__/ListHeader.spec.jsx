import React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import ListHeader from '../ListHeader';

describe('<ListDelete>', () => {
  let component;

  beforeEach(() => {
    const props = {
      boardId: 'BoardID',
      listId: 'ListID',
      listTitle: 'List Title',
    };

    component = shallow(<ListHeader {...props} />);
  });

  it('should render with given state from Redux store', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('checks if toggleOpened sets isOpened correctly', () => {
    component.instance().toggleOpened();
    expect(component.instance().state.isOpened).toEqual(true);

    component.instance().toggleOpened();
    expect(component.instance().state.isOpened).toEqual(false);
  });

  it("should call toggleOpened after click on icon", () => {
    const spy = jest.spyOn(component.instance(), "toggleOpened");

    component.find('Icon').simulate('click');

    expect(spy).toHaveBeenCalledTimes(1);
  });

  it('should render ListActions if isOpened is true', () => {
    component.instance().toggleOpened();

    expect(toJson(component)).toMatchSnapshot();
  });
});
