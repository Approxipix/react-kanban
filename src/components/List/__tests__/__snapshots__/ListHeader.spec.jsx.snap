// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListDelete> should render ListActions if isOpened is true 1`] = `
<Fragment>
  <styled.div>
    <ListTitle
      listId="ListID"
      listTitle="List Title"
    />
    <Icon
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="ellipsis-h"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </Icon>
  </styled.div>
  <ListActions
    boardId="BoardID"
    listId="ListID"
    toggleOpened={[Function]}
  />
</Fragment>
`;

exports[`<ListDelete> should render with given state from Redux store 1`] = `
<Fragment>
  <styled.div>
    <ListTitle
      listId="ListID"
      listTitle="List Title"
    />
    <Icon
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="ellipsis-h"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </Icon>
  </styled.div>
</Fragment>
`;
